system/jlib/jthread.cpp

```diff
@@ -725,18 +719,13 @@ void CAsyncFor::For(unsigned num,unsigned maxatonce,bool abortFollowingException
                     }
                     catch (IException * _e)
                     {
-                        synchronized block(*errmutex);
-                        if (erre)
-                            _e->Release();  // only return first
-                        else
-                            erre = _e;
+                        erre.setownIfNull(_e);
                     }
     #ifndef NO_CATCHALL
                     catch (...)
                     {
-                        synchronized block(*errmutex);
-                        if (!erre)
-                            erre = MakeStringException(0, "Unknown exception in Thread %s", getName());
+                        if (!erre.isSet())
```

Comment:
Doesn't the ifNull imply only if set already?

Reply:
I think it's checking, to avoid the MakeStringException call/creation, which might be thrown away if erre is already.
Might not be worth saving the wasted creating and just call setownIfNull(makeStringException....) for clearer code?

Reply:
That's how I had it to start off with.  I'll change it back.